### PR TITLE
WIP Push notices issue #25

### PR DESCRIPTION
**Pull request by adamrais**
Added notifications section in the settings page

**What is done:**
Added the notifications in the setting section.
Toggle buttons for each notifications
logic for when shops open / close

**What is left to do:**
Logic for  
- special events (sprint fishing tourney)
- turnip price changes
- turnip sell reminder / turnip purchase reminder

Let me know for the rest

### CODE REVIEW

**Comment by TheVaan** [comment]
Why are there 59 deletions of assets in the commit? Please revert them. 
Could you take a look on existing notifications (`registerTurnipsPredictionNotification`) and adopt handling of text so we get localizable notification „out of the box“?

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift [hunk 1]
```diff
@@ -21,12 +21,13 @@ struct SettingsView: View {
     @State private var importedFile: URL?
     @State private var showSuccessImportAlert = false
     @State private var showDeleteConfirmationAlert = false
-
+
```
Comment (TheVaan) [inline ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift]:
```suggestion

```

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift [hunk 2]
```diff
@@ -151,6 +159,46 @@ struct SettingsView: View {
...
+                }
+                .onTapGesture {
+                    if !self.appUserDefaults.isShopOpenClose {
+                        self.getNotify(title: "ACHNBrowserUI", subtitle: "Nook’s Cranny store is open", hour: 8, minute: 0, isRepeated: true)
+                        self.getNotify(title: "ACHNBrowserUI", subtitle: "Nook’s Cranny store is closing in 1h", hour: 21, minute: 0, isRepeated: true)
+                        self.getNotify(title: "ACHNBrowserUI", subtitle: "Able Sisters Shop is open", hour: 9, minute: 0, isRepeated: true)
+                        self.getNotify(title: "ACHNBrowserUI", subtitle: "Able Sisters Shop is closing in 1h", hour: 20, minute: 0, isRepeated: true)
+                        
+                        // for debugging
+                        print("toggled")
```
Comment (TheVaan) [inline ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift]: Remove debugging output

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift [hunk 3]
```diff
@@ -205,6 +253,35 @@ struct SettingsView: View {
                     content: { self.deleteConfirmationAlert })
         }
     }
+    
+    func getNotify(title: String, subtitle: String, hour: Int, minute: Int, isRepeated: Bool) {
```
Comment (TheVaan) [inline ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift]: Don’t do this here. Add notifications to `NotificationsManager` in Backend package.

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift [hunk 4]
```diff
@@ -151,6 +159,46 @@ struct SettingsView: View {
...
     }
     
+    private var notificationSection: some View {
+        Section(header: SectionHeaderView(text: "Notifications", icon: "clock")) {
+            HStack {
+                Toggle(isOn: $appUserDefaults.isShopOpenClose ) {
+                    Text("When shops open/close")
+                }
+                .onTapGesture {
+                    if !self.appUserDefaults.isShopOpenClose {
```
Comment (TheVaan) [inline ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift]: Don’t register notifications in settings view. This belongs to `NotificationsManager` in the Backend package

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift [hunk 5]
```diff
@@ -151,6 +159,46 @@ struct SettingsView: View {
...
+                        self.getNotify(title: "ACHNBrowserUI", subtitle: "Nook’s Cranny store is open", hour: 8, minute: 0, isRepeated: true)
```
Comment (TheVaan) [inline ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift]: Title of the notification should not be the repository name. If you want to use the name of the app, use „AC Helper“

File: ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift (hunk 4 above)
Comment (adamrais) [reply ACHNBrowserUI/ACHNBrowserUI/views/settings/SettingsView.swift:168]: Do you know how I should call NotificationsManager? I cannot use an EnvironmentObject

**Comment by rohanrk** [comment]
So it looks like we have separate PRs on the same issue. Never got a follow up but do you want to have a chat on merging these into one and dividing the work accordingly?

**Comment by adamrais** [comment]
> So it looks like we have separate PRs on the same issue. Never got a follow up but do you want to have a chat on merging these into one and dividing the work accordingly?

Hello, sure do you have a preferred way of communication?

**Comment by rohanrk** [comment]
Email's preferred. rohanrk2009@gmail.com is my address

**Pull request closed without merging**